core/controllers/acl_decorators.py

```diff
@@ -151,10 +151,8 @@ def test_does_classroom_exist(self, classroom_url_fragment, **kwargs):
             classroom_url_fragment)
 
         if not classroom:
-            _redirect_based_on_return_type(
-                self, '/learn/%s' % constants.DEFAULT_CLASSROOM_URL_FRAGMENT,
-                self.GET_HANDLER_ERROR_RETURN_TYPE)
-            return
+            if self.GET_HANDLER_ERROR_RETURN_TYPE == feconf.HANDLER_TYPE_JSON:
```

Comment:
I think this needs an explanation, or an 'else' explaining what we do if it's not a json handler.

Reply:
comment added